flask_restful/reqparse.py

```diff
@@ -122,6 +122,8 @@ def source(self, request):
         else:
             values = MultiDict()
             for l in self.location:
+                if not request.is_json and l == 'json':
```

Comment:
Switching the order of these checks is probably more efficient.

Reply:
Granted, re: your comment on the issue, this may be moot if maintenance has dropped off here, and we may need to look more closely at other options.